### ISSUE TITLE
no-extra-parens has a false negative in the left side of for loops

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#8393]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.8.0
* **npm Version:** 4.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml
rules:
  no-extra-parens: error
parserOptions:
  ecmaVersion: 2015
```

**What did you do? Please include the actual source code causing the issue.**

```js
for ((foo) in bar);
for ((foo) of bar);
```

**What did you expect to happen?**

I expected errors to be reported for both cases.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors were reported.

The fix for this should make sure to handle spacing correctly. For example, a space needs to be inserted after `foo` here when autofixing to avoid a syntax error:

```js
for ((foo)in bar);
```

**Comment by VictorHom** [comment]
@not-an-aardvark can I help out with this?

**Comment by not-an-aardvark** [comment]
Sure, feel free!

**Comment by VictorHom** [comment]
@not-an-aardvark I have a pr for this issue: https://github.com/eslint/eslint/pull/8679

When you have some time, could you review?
